src/Tool/MacAuthorizationTrait.php

```diff
@@ -26,7 +26,7 @@
     /**
      * Returns the id of this token for MAC generation.
      *
-     * @param  AccessToken $token
+     * @param  AccessToken|null $token
```

Comment:
This stems from this Scrutinizer message:

> It seems like `$token` defined by parameter `$token` on line 65 can be `null`; however, `League\OAuth2\Client\Tool\MacAuthorizationTrait::getTokenId()` does not accept `null`, maybe add an additional type check?

The signature for the method in which this suggestion occurs is:

    protected function getAuthorizationHeaders($token = null)

This method calls `getTokenId()`, which doesn't allow a `null` token to be passed in (according to its return type annotation). Perhaps the solution is to not default to `null` for `getAuthorizationHeaders()`.